tests/Mawosoft.PowerShell.WindowsSearchManager.Tests/AddSearchRootCommandTests.cs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2023 Matthias Wolf, Mawosoft.
+
+namespace Mawosoft.PowerShell.WindowsSearchManager.Tests;
+
+public class AddSearchRootCommandTests : CommandTestBase
+{
+    [Theory]
+    [InlineData(@"Add-SearchRoot -Path x:\foo", @"x:\foo")]
+    [InlineData(@"Add-SearchRoot -Path x:\foo, x:\bar", @"x:\foo", @"x:\bar")]
+    [InlineData(@"Add-SearchRoot x:\foo, x:\bar", @"x:\foo", @"x:\bar")]
+    [InlineData(@"@('x:\foo', 'x:\bar') | Add-SearchRoot", @"x:\foo", @"x:\bar")]
+    public void PathParameterSet_Succeeds(string script, params string[] expectedMethodParameters)
+    {
+        Collection<PSObject> results = InvokeScript(script);
+        Assert.Empty(results);
+        Assert.False(PowerShell.HadErrors);
+        var recordedCallInfos = InterfaceChain.ScopeManager.RecordedCallInfos;
+        Assert.All(recordedCallInfos.Take(recordedCallInfos.Count - 1),
+            (item, i) =>
+            {
+                Assert.Equal("AddRoot", item.MethodName);
+                ISearchRoot root = (Assert.Single(item.Parameters) as ISearchRoot)!;
+                Assert.NotNull(root);
+                SearchRootInfo expected = new() { Path = expectedMethodParameters[i] };
+                SearchRootInfo actual = new(root);
+                Assert.Equivalent(expected, actual, strict: true);
+            });
+        Assert.Equal(expectedMethodParameters.Length, recordedCallInfos.Count - 1);
+        Assert.Equal("SaveAll()", InterfaceChain.ScopeManager.RecordedCalls[recordedCallInfos.Count - 1]);
+        Assert.True(InterfaceChain.SingleHasWriteRecordings(InterfaceChain.ScopeManager));
+    }
+
+    private static readonly List<SearchRootInfo> s_rootInfos = new()
+    {
+        new()
+        {
+            Path = @"fooprotocol://{bar-sid}/",
+            ProvidesNotifications = false,
+            UseNotificationsOnly = true,
+            EnumerationDepth = 333,
+            HostDepth = 222,
+            AuthenticationType = _AUTH_TYPE.eAUTH_TYPE_BASIC
+        },
+        new()
+        {
+            Path = @"file:///c:\",
+            IsHierarchical = false,
+            UseNotificationsOnly = true,
+            EnumerationDepth = 333,
+            HostDepth = 222,
+        },
+        new()
+        {
+            Path = @"bar://foo/",
+            FollowDirectories = false,
+            AuthenticationType = _AUTH_TYPE.eAUTH_TYPE_NTLM
+        },
+        new()
+        {
+            Path = @"x:\",
+        },
+        new()
+        {
+            Path = @"y:\",
+        }
+    };
+
+    [Theory]
+    [InlineData(1, typeof(SearchRootInfo), false)]
+    [InlineData(1, typeof(PSObject), false)]
+    [InlineData(1, typeof(PSCustomObject), false)]
+    [InlineData(-1, typeof(SearchRootInfo), false)]
+    [InlineData(-1, typeof(PSObject), false)]
+    [InlineData(-1, typeof(PSCustomObject), false)]
+    [InlineData(-1, typeof(SearchRootInfo), true)]
+    [InlineData(-1, typeof(PSObject), true)]
```

Comment:
Use nameof(type) instead of typeof(type). This shortens the displayed parameter list in test results and allows using switch instead of if... else if.